### ISSUE TITLE
References do not show up in search results anymore when searching for their text content

### ISSUE THREAD

**Issue by hannesfrank** [remnoteio/remnote-issues#770]
![image](https://user-images.githubusercontent.com/4137110/107418527-107d5080-6b17-11eb-9f5d-458eb2d78397.png)

I'd expect the second and the fifth rem to show as search results as well. I think they used to show up before? Or is this working as intended?

Windows, Desktop App 1.1.12.

**Comment by mfranzs** [comment]
This is currently working as intended - if the search result text **only** appears within a Rem Reference, that Rem will not be matched. However, if text outside of the Rem Reference matches, then bonus points will be awarded for also matching the text within the Rem Reference. Is this breaking a workflow for you?

**Comment by badjerphoenix** [comment]
I would agree with @hannesfrank that my expectation would be for the all mentions of the search item to show up. In this case that includes 2nd & 5th rem. This should include *text* & *text within reference*, not just the *text search*.

Breaks workflow as need to *jump to backlinks from search* for reference text as a middle man. Would rather see all mentions, click from the search & be jumped to the given text rem/ reference rem. Increases fluidity as less jumping around. Also allows us to see if we have inadvertently used same naming for different rem

**Comment by badjerphoenix** [comment]
*Text within reference* seems to populate with 1st term now. Presume it is a *or/ and* statement as when I use 2nd term it jumps from 5 to 103

Are stubs within search supposed to function same as the *text* & the *text within references* via search?

**Comment by Add-HX** [comment]
I am also encountering this problem. I have a Concept called "Orthogonal Matrix" that consists of two Rem references to Orthogonal and Matrix, but when I search the full world, it disappears from the top of the search bar. Searching "Orthogonal Mat" would put it close to the top, but searching "Orthogonal Matrix" makes it disappear from all search results.

**Comment by ognsya** [comment]
As @badjerphoenix mentioned, this seems to be more due to limitation of search, which does not accept more than one word (not even using ""). Hopefully this is added soon, as this is a very essential search feature.

Limited search is one of the main reasons I (still) can't migrate most of my workflow to RemNote from Notepad++ (!).

**Comment by badjerphoenix** [comment]
> _Text within reference_ seems to populate with 1st term now

Seems that the issue has now re-occurred. Was working with criteria listed in link to quote

Now when I enter the 1st character is the only time that both *rem text & rem reference* pulls items in based upon search item.
Once I type more than once character (within the same term) it reverts back to *only* pulling up items that are *text rem* as the *reference rem will not be pulled in*

IE in example *gertrude* is search item

![image](https://user-images.githubusercontent.com/78933852/107796946-fcf10600-6d1f-11eb-93a7-faccbc1b5e85.png)

![image](https://user-images.githubusercontent.com/78933852/107797014-12663000-6d20-11eb-9938-558f6bb47c47.png)

**Comment by badjerphoenix** [comment]
![image](https://user-images.githubusercontent.com/78933852/107797152-3aee2a00-6d20-11eb-8914-9e47d111745c.png)

**Comment by hannesfrank** [comment]
@badjerphoenix The `g` matches the `G` in Sandbox Testing **G**rounds. It has nothing to do with only working for a single character.

I slightly disagree that Ctrl + P search should help you discover all mentions. At least for `[[` and Ctlr + P you are looking for a single rem and the objective of a smart search is to always just magically find the exact rem you are looking. It's true though that there should be a low friction method to explore if you do _not_ know what you are looking for. And the Ctrl + P search UX is pretty smooth and might have the potential to do so.

The discovery UX to search through all rem mentioning something should probably be a mix of a search portal/query and Ctrl + P.
- It should be somewhat persistent and not a popup to being able to pull multiple search results into the surrounding page.
- It should show more context on hover to know what you are looking at.
- It should be as fast, interactive and visually pleasing as the Ctrl + P search.
- It should be powerful and configurable to tweak the search results.

---

@mfranzs I personally don't have a workflow that requires this explicitly, but there are some rough edges I am not 100% sure how to model things because I'm not an expert with the references + outliner combo either.

For example Nested References like `[[Habit]] Tracker` would require matching references in search. There are a few ways to model this, but they all have their disadvantages:
0. The top level `[[Habit]] Tracker` would not show up in search until you typed `Habit T`.
1. Make it a child of `Habit`: `Habit > Tracker`. But then the reference is just `| Tracker`, Both the extra | and the missing Habit is bad.
    - And there is no way to set reference style to always full name by default on a per rem basis.
2. You could `Habit > Habit Tracker` but that is redundant and makes renaming harder. This is my current approach because it looks cleanest.
3 `Habit > [[Habit]] Tracker` and e.g. make `[[Habit]] Tracker` a document to open it directly when searching for it.

More flexibility/configurability is always good I think:
- Configurable serach parameters: https://forum.remnote.io/t/settings-search-parameters-weights/2647
- Configurable reference styles (show reference as full name by default by tagging original with power up, more data tags for Custom CSS)
- Configurable search contexts (e.g. exclude subtrees/terms from search etc.)

---

**I think that you already enabled matching inside references again?**

![image](https://user-images.githubusercontent.com/4137110/107924393-2786cd00-6f73-11eb-8c00-6c8d3c5836ec.png)

**Comment by badjerphoenix** [comment]
> The g matches the G in Sandbox Testing Grounds. It has nothing to do with only working for a single character
 
 I did not realize that it was not pulling it in based on the `G` within the reference itself but only due to the `G` that is within the breadcrumb trail. I made the presumption it was based on the reference as well due to the blue `G` being highlighted in conjunction with the breadcrumb trail.
 
 > disagree that Ctrl + P search should help you discover all mentions. At least for [[ and Ctlr + P you are looking for a single rem and the objective of a smart search is to always just magically find the exact rem you are looking.
 
 If this is the intended effect, that only the original rem is the desired search result (& not all mentions), then I have misunderstood it's functionality within the app & the resulting increase of friction in my workflow. I am not wanting to increase the jump pts required to get to the backlinks. Will think on the matter outside of GitHub then as it sounds like this is the intended function from the app perspective.
